parser-typechecker/src/Unison/TypePrinter.hs

```diff
@@ -45,42 +46,43 @@ import qualified Unison.DataDeclaration as DD
 -}
 
 prettyTop
-  :: forall s v a . (IsString s, LL.ListLike s Char, Var v)
-  => PrettyPrintEnv -> AnnotatedType v a -> Pretty s
+  :: forall v a . (Var v)
+  => PrettyPrintEnv -> AnnotatedType v a -> Pretty ColorText
 prettyTop ppe ty = pretty ppe mempty (-1) ty
 
 pretty
-  :: forall s v a . (IsString s, LL.ListLike s Char, Var v)
+  :: forall v a . (Var v)
   => PrettyPrintEnv
   -> Imports
   -> Int
   -> AnnotatedType v a
-  -> Pretty s
+  -> Pretty ColorText
```

Comment:
I'm thinking we should work with `Pretty (AnnotatedText SyntaxHighlighting.Element)`, and then `Pretty.map` is used to convert that to `Pretty ColorText` just before rendering to ANSI. For rendering to ANSI terminal you may not use all the colors given how limited the palette is, but it's unfortunate to throw that information away because other targets could use it.

For instance, if you're rendering a `Pretty (AnnotatedText Element)` to HTML or some Javascript-y widget, you'd likely have a CSS class for each `Element` so that the whole thing can be styled with CSS.

Reply:
Ah, good point!! I will make this change, but it will probably take me a week or so to get round to it. I'll still do it even if you merge the PR in the mean time.

Reply:
@pchiusano umm, I just banged the following commit in without asking, doing a couple of renames, hope that's not too much of a transgression given it's trivial - I wanted to minimize any merging.  Remainder of this refactor will get a PR as per proper procedure.  Shout if I should undo/change the renames.

https://github.com/unisonweb/unison/commit/45122cb4a019a99f8afa8f4ca5c7057ef1983b1e

Reply:
(the rest of the refactor is in https://github.com/unisonweb/unison/pull/564)